ui/app/components/app/menu-droppo.js

```diff
@@ -1,33 +1,92 @@
+import PropTypes from 'prop-types'
 import React, { Component } from 'react'
-import { inherits } from 'util'
 import { findDOMNode } from 'react-dom'
 import ReactCSSTransitionGroup from 'react-transition-group/CSSTransitionGroup'
 
-export default MenuDroppoComponent
+export default class MenuDroppoComponent extends Component {
+  static propTypes = {
+    isOpen: PropTypes.bool.isRequired,
+    innerStyle: PropTypes.object,
+    children: PropTypes.node.isRequired,
+    onClickOutside: PropTypes.func.isRequired,
+    containerClassName: PropTypes.string,
+    zIndex: PropTypes.number,
+    style: PropTypes.object.isRequired,
+    useCssTransition: PropTypes.bool.isRequired,
+    speed: PropTypes.string,
+  }
 
+  renderPrimary () {
+    const isOpen = this.props.isOpen
+    if (!isOpen) {
+      return null
+    }
 
-inherits(MenuDroppoComponent, Component)
-function MenuDroppoComponent () {
-  Component.call(this)
-}
+    const innerStyle = this.props.innerStyle || {}
+
+    return (
+      <div className="menu-droppo" key="menu-droppo-drawer" style={innerStyle}>
+        {this.props.children}
+      </div>
+    )
+  }
+
+  manageListeners () {
+    const isOpen = this.props.isOpen
+    const onClickOutside = this.props.onClickOutside
+
+    if (isOpen) {
+      this.outsideClickHandler = onClickOutside
+    } else if (isOpen) {
+      this.outsideClickHandler = null
+    }
+  }
+
+  componentDidMount = function () {
+    if (this && document.body) {
```

Comment:
Well, this is a weird thing to check for

Reply:
Yep